### DISCUSSION TITLE
Does FreshRSS change a footnote's href?

### DISCUSSION THREAD

**Discussion by vincode-io** [FreshRSS/FreshRSS#4476] [Q&A]
I've noticed that NetNewsWire footnote detection code doesn't work when using a FreshRSS account. It looks like the normal anchor link in the footnote gets changed to a fully resolved link. Is this intended behavior or is something else going on?

**Comment by Alkarex** [comment]
Could you please post a screenshot or two to illustrate the issue?

**Reply by vincode-io** [reply]
Do you want a screenshot of the HTML? Maybe I didn't explain my question well enough.

Footnote hrefs contain an anchor link to another location in the page. This anchor link starts with "#fn". This is what the raw RSS content has in it. 

The HTML that FreshRSS delivers appears to change the original href to be a fully resolved link. This link points to the original article on the web instead of the page anchor.

The net effect of this is that the NetNewsWire footnote code doesn't recognize it as a footnote. It is now just a regular link that takes the user out of NetNewsWire instead of displaying the footnote inline.

I'm interested in the rationale behind changing the footnote hrefs to fully resolved URL's.

**Reply by Krinkle** [reply]
More details at https://github.com/Ranchero-Software/NetNewsWire/issues/3564#issuecomment-1182073331

> Original:
> ```html
> <content type="html" xml:base="https://timotijhof.net/posts/2022/internet-archive-crawling/">
>   <p>..<sup ..><a href="#fn:sse" class="footnote" rel="footnote">1</a></sup> ..
> ```
> 
> FreshRSS proxy:
> ```html
> <description>
>   <p>.<sup ..><a href="https://timotijhof.net/posts/2022/internet-archive-crawling/#fn:sse" rel="footnote">1</a></sup> ..
> ```

**Reply by Alkarex** [reply]
Thanks for the code snippet. Now I understand. Yes indeed, relative URLs are resolved by the underlying SimplePie library. We could try to catch such anchors though.

**Reply by Alkarex** [reply]
https://github.com/FreshRSS/FreshRSS/issues/4481

**Reply by vincode-io** [reply]
Thanks for the follow up @Alkarex!
